scripts/ai/AIDriveStrategyCourse.lua

```diff
@@ -278,4 +278,13 @@ end
 --- Are we within distance meters of the first waypoint (measured on the course, not direct path)?
 function AIDriveStrategyCourse:isCloseToCourseStart(distance)
     return self.course:getDistanceFromFirstWaypoint(self.ppc:getCurrentWaypointIx()) < distance
-end
\ No newline at end of file
+end
+
+function AIDriveStrategyCourse:delete()
+    if self.course then 
+        --- Remembers the third last waypoint ix driven.
+        local ix = math.max(1,self.course:getCurrentWaypointIx()-2)

```

Comment:
We have function for this: rememberWaypointToContinueFieldwork() as it isn't trivial